developer/js/source/model-info-compiler/model-info-compiler.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Merges a source .model_info file with metadata extracted from .kps file and
+ * compiled files to produce a comprehensive .model_info file.
+ */
+
+/// <reference path="../lexical-model-compiler/lexical-model.ts" />
+/// <reference path="../lexical-model-compiler/model-info-file.ts" />
+
+import * as fs from "fs";
+import * as path from "path";
+
+const minKeymanVersion = '12.0';
+
+export default class ModelInfoCompiler {
+
+/**
+   * Merges source .model_info file with metadata from the model and package source file.
+   * This function is intended for use within the lexical-models repository. While many of the
+   * parameters could be deduced from each other, they are specified here to reduce the 
+   * number of places the filenames are constructed.
+   *
+   * @param sourceModelInfoFileName  Path for the source .model_info file
+   * @param destModelInfoFileName    Path to write the merged .model_info file to
+   * @param model_id                 The identifier for the model
+   * @param kmpJsonData              The data from the .kps file, transformed to kmp.json
+   * @param sourcePath               The path in the keymanapp/lexical-models repo where this model may be found
+   * @param modelFileName            The compiled model filename and relative path (.js)
+   * @param kmpFileName              The compiled package filename and relative path (.kmp)
+   */
+  writeMergedModelMetadataFile(
+    sourceModelInfoFileName: string,
+    destModelInfoFileName: string, 
+    model_id: string, 
+    kmpJsonData: KmpJsonFile, 
+    sourcePath: string|null|undefined, 
+    modelFileName: string, 
+    kmpFileName: string
+  ) {
+    
+    /*
+     * Model info looks like this:
+     *
+     *  {
+     *    "name": "Example Template Model"
+     *    "license": "mit",
+     *    "version": "1.0.0",
+     *    "languages": ["en"],
+     *    "authorName": "Example Author",
+     *    "authorEmail": "nobody@example.com",
+     *    "description": "Example wordlist model"
+     *  }
+     *
+     * For full documentation, see:
+     * https://help.keyman.com/developer/cloud/model_info/1.0/
+     */
+    let model_info: ModelInfoFile = JSON.parse(fs.readFileSync(sourceModelInfoFileName, 'utf8'));
+
+    //
+    // Build merged .model_info file
+    // https://api.keyman.com/schemas/model_info.source.json and
+    // https://api.keyman.com/schemas/model_info.distribution.json
+    // https://help.keyman.com/developer/cloud/model_info/1.0
+    //
+
+    function set_model_metadata(field: string, expected: any, warn: boolean = true) {
```

Comment:
Oh! Here's more mixed `snake_case` and `camelCase`!

Reply:
![image](https://user-images.githubusercontent.com/4498365/64083409-0e7c6300-cd63-11e9-99c8-12239814c824.png)